Release Note v11.1.0 (#228)
mia-platform/documentation: docs/release-notes/v11.1.0.md

#### "Undefined" in the detail of a Pod error

We fixed an issue that prevented the user to understand the type of error running in a Pod, by showing a generic "undefined" in the information popup.

## Fast Data

### New Features

### Improvements

### Bug Fix

## Marketplace

### New Marketplace Components

#### OpenAI Playground

This is a Next.js with React example in preview that allows you to try-out OpenAI APIs directly in Console. This service will help you leverage AI potentialities to answer any question you need to be answered.  
You can find more info [here](https://github.com/mia-platform-marketplace/open-ai-showcase).

### Marketplace Updates

### New Libraries

We are excited to announce the release of a new open-source library written in TypeScript that will help you easily check Kafka probes in your microservices: [Kafka HealthChecker](https://github.com/mia-platform/kafka-healthchecker). This library is designed to simplify the process of monitoring Kafka by accepting in input a list of producers and consumers and providing two methods that will check if your service is connected to Kafka and able to process new messages.

> We have included detailed [documentation](https://docs.mia-platform.eu/docs/runtime_suite_libraries/kafka-healthchecker/overview) to help you get up and running with the library as quickly as possible.

Author: ```suggestion
We have included detailed [documentation](../runtime_suite_libraries/kafka-healthchecker/overview.md) to help you get up and running with the library as quickly as possible.
```

---
id: v11.1.0
title: Version v11.1.0 Release Notes
sidebar_label: v11.1.0
image: "img/release-note-link-preview.png"
---

_May 25th, 2023_

## Console

### New Features

> #### Group of Users in Console

Reviewer: ```suggestion
#### Users Group Management
```

> 
> From this Console release, Company Owners will be able to manage Company users also through groups.  
> Groups are a new identity type aimed at simplifying the management of a multitude of users who have to be assigned the same privileges (e.g. a work team).  
> Management of users through groups sits alongside the actual individual management of users, to allow you to better structure access to resources as you prefer, giving you the freedom to choose how to govern access levels of your Company users. 
> 
> To discover more about groups of users and how they work, just take a look at the [groups documentation page](../development_suite/identity-and-access-management/group-management.md). 

Reviewer: ```suggestion

From a dedicated section inside of the Company Overview, Company Owners can now manage Company users also through Groups.  
Thanks to Groups it is possible to simplify the management of a multitude of users who have to be assigned the same privileges (e.g. a work team).  
Management of users through groups sits alongside the actual individual management of users, to allow you to better structure access to resources as you prefer, giving you the freedom to choose how to govern access levels of your Company users.  

To discover more about user groups management, just take a look at the related [documentation page](../development_suite/identity-and-access-management/group-management.md). 
```

* An error that caused the Debug Area of the Console to fail to load has been fixed
* A bug in the Project creation that caused a malfunction in the display of the default CI/CD tool of each environment has been fixed
* An issue that prevented the user to understand the type of error running in a Pod, by showing a generic "undefined" in the information popup has been fixed
* The Backoffice Configurator now correctly configures the `oneOf` properties, handling it properly in editing modal

## Fast Data

### New Features

#### Ingestion Reloader

Introducing a new version of the Ingestion Reloader, the `1.2.0`! With this new verison, you can now filter specific messages retrieved from a bucket, using a custom selector system and allowing precise and targeted filtering of messages to be reingested: just specify the file with it in the `customSelectorName` field. The selector's configurations are available at the POST `/reingestion/file` and POST `/reingestion/topic` endpoints.

Go to the [official documentation](/fast_data/bucket_storage_support/configuration/ingestion_reloader.md) to find out more.

#### Debezium DB2 plugin

> The rollout of the supported Debezium Plugins is going on! You can now find in the Marketplace the [Debezium connector plugin for DB2](/fast_data/connectors/debezium_cdc#db2.md) databases with the pre-configured config map for Debezium Server to connect directly to a DB2 Database.

Reviewer 2: @giuliowaitforitdavide is it a template, not a plugin, am I right?

Reviewer 3: Fixed

Management of users through groups sits alongside the actual individual management of users, to allow you to better structure access to resources as you prefer, giving you the freedom to choose how to govern access levels of your Company users.  

To discover more about user groups management, just take a look at the related [documentation page](../development_suite/identity-and-access-management/group-management.md). 

# Privatized Marketplace for Companies

From now on, Companies can have their own private Marketplace items, which will not be visible and usable from those who are not part of those specific Companies.   
More specifically, Company Owners and Project Administrators can now be able to manage – directly from CMS – Marketplace resources that will be available exclusively for the members of their Company. This has to be done 
by specifying the `tenantId` as a Company identifier. 

To discover more information, visit the dedicated Marketplace [documentation page](../marketplace/add_to_marketplace/contributing_overview#public-and-private-marketplace.md).

### Improvements

> ### Bug Fix

Reviewer 2: @marcofilippi @albertotessarotto are we doing again the paragraph for each bug fixes? Why don't we write the list as in the past release notes?

I need to add a bugfix for the backoffice configurator, I need to understand where to put it

Author: agree, we now modify the structure of the Bugfix section to make it the same as in the previous rn

Author: I modified, give a check if it is okay for you @TheRusCompass

#### Error displaying Debug section

A bug that caused the Debug Area of the Console to fail to load has been fixed.

#### CI/CD tool malfunction in project creation

It has been fixed a bug in the Project creation that caused a malfunction in the display of the default CI/CD tool of each environment.

#### "Undefined" in the detail of a Pod error

We fixed an issue that prevented the user to understand the type of error running in a Pod, by showing a generic "undefined" in the information popup.

## Fast Data

### New Features

#### Ingestion Reloader

With the new version of the Ingestion Reloader, the `1.2.0`, you can now filter specific messages retrieved from a bucket, using a custom selector system and allowing precise and targeted filtering of messages to be reingested: you just need to specify the file with it in the `customSelectorName` field. The selector's configurations are available at the POST `/reingestion/file` and POST `/reingestion/topic` endpoints.

#### Debezium DB2 plugin

> You can now find in the Marketplace the [Debezium connector plugin for DB2](/documentation/docs/fast_data/connectors/debezium_cdc#db2.md) databases.

Reviewer 2: @giuliowaitforitdavide let's try to write similar stuff of the past release notes

Reviewer 3: Done, let me know if it's ok

---
id: v11.1.0
title: Version v11.1.0 Release Notes
sidebar_label: v11.1.0
image: "img/release-note-link-preview.png"
---

_May 25th, 2023_

## Console

### New Features

#### User Groups Management

From a dedicated section inside of the Company Overview, Company Owners can now manage Company users also through Groups.  

> Thanks to Groups it is possible to simplify the management of a multitude of users who have to be assigned the same privileges (e.g. a work team).  

Author: ```suggestion
Thanks to groups it is possible to simplify and speed-up the management of a multitude of users who have to be assigned the same privileges.  
```

More specifically, Company Owners and Project Administrators can now be able to manage – directly from CMS – Marketplace resources that will be available exclusively for the members of their Company. 
This has to be done by specifying the `tenantId` as a Company identifier. 

To discover more information, visit the dedicated Marketplace [documentation page](../marketplace/add_to_marketplace/contributing_overview#public-and-private-marketplace.md).

### Improvements

#### Check of Certificate Authority in cluster connection

In Cluster creation, adding the Certificate Authority is actually not mandatory but it depends on the properties of each Cluster. It will always be checked: if the cluster needs a CA it will fail the connection if it has not been entered.

### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

* An error that caused the Debug Area of the Console to fail to load has been fixed;
* A bug in the Project creation that caused a malfunction in the display of the default CI/CD tool of each environment has been fixed;

> * A bug that caused a malfunction of the multi-log functionality of pods in the runtime area has been fixed;

Reviewer: ```suggestion
* A bug that caused a malfunction of the multi-log functionality of pods in the Runtime area has been fixed;
```

> * An issue that prevented the user to understand the type of error running in a Pod, by showing a generic "undefined" in the information popup has been fixed;

Reviewer: ```suggestion
* An issue that prevented the user to understand the type of error running in a Pod, by showing a generic "undefined" in the information popup, has been fixed;
```

> ### Bug Fix
> 
> This version addressed a few bugs, here is a comprehensive list:
> 
> * An error that caused the Debug Area of the Console to fail to load has been fixed;
> * A bug in the Project creation that caused a malfunction in the display of the default CI/CD tool of each environment has been fixed;
> * A bug that caused a malfunction of the multi-log functionality of pods in the runtime area has been fixed;
> * An issue that prevented the user to understand the type of error running in a Pod, by showing a generic "undefined" in the information popup has been fixed;
> * The Backoffice Configurator now correctly configures the `oneOf` properties, handling it properly in editing modal.

Reviewer: ```suggestion
### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

* An error that caused the Debug Area of the Console to fail to load has been fixed;
* A bug in the Project creation that caused a malfunction in the display of the default CI/CD tool of each environment has been fixed;
* We solved an error that caused a malfunction of the multi-log functionality of Pods in the Runtime area;
* An issue that prevented the user to understand the type of error running in a Pod, by showing a generic "undefined" in the information popup, has been fixed;
* The Backoffice Configurator now correctly configures the `oneOf` properties, handling it properly in editing modal.
```

---
id: v11.1.0
title: Version v11.1.0 Release Notes
sidebar_label: v11.1.0
image: "img/release-note-link-preview.png"
---

_May 25th, 2023_

## Console

### New Features

#### User Groups Management

From a dedicated section inside of the Company Overview, Company Owners can now manage Company users also through Groups.  
Thanks to groups it is possible to simplify and speed-up the management of a multitude of users who have to be assigned the same privileges.  
Management of users through groups sits alongside the actual individual management of users, to allow you to better structure access to resources as you prefer, giving you the freedom to choose how to govern access levels of your Company users.  

![Groups section](./img/overview/group_table.png)

To discover more about user groups management, just take a look at the related [documentation page](../development_suite/identity-and-access-management/group-management.md). 

> # Privatized Marketplace for Companies

Author: ```suggestion
## Privatized Marketplace for Companies
```

> ## Privatized Marketplace for Companies

Author: ```suggestion
#### Privatized Marketplace for Companies
```

> From now on, Companies can have their own private Marketplace items, which will not be visible and usable from those who are not part of those specific Companies.   
> More specifically, Company Owners and Project Administrators can now be able to manage – directly from CMS – Marketplace resources that will be available exclusively for the members of their Company. 
> This has to be done by specifying the `tenantId` as a Company identifier. 
> 
> To discover more information, visit the dedicated Marketplace [documentation page](../marketplace/add_to_marketplace/contributing_overview#public-and-private-marketplace.md).

Reviewer: ```suggestion
From now on, Companies can have their own private Marketplace items, which will not be visible and usable from those who are not part of those specific Companies.   
More specifically, Company Owners and Project Administrators can now be able to manage – directly from CMS – Marketplace resources that will be available exclusively for the members of their Company. 
This has to be done by specifying the Company identifier field.

To discover more information, visit the dedicated Marketplace [documentation page](../marketplace/add_to_marketplace/contributing_overview#public-and-private-marketplace.md).
```

#### `bk-card` supports horizontal layout for footer buttons
`bk-card` allows to place footer buttons horizontally through key `buttonsLayout` in `cardSchema.footer`

#### Logo can be fetched form current user information
`bk-layout` supports retrieving logo URL from user data, which is retrieved from a configurable endpoint through an http call

#### Theming variables can be retrieved with http call
`bk-antd-theme-manager` supports retrieving theming information from a configurable endpoint through an http call

### Bug Fix

#### Buttons in `bk-confirmation-modal` have headers and credentials
`bk-confirmation-modal` injects `headers` and `credentials` into dynamically mounted buttons

#### Http `bk-buttons` support `loadingOnAction` and `disableOnAction`
`bk-button`s with action of type `http` correctly implement `loadingOnAction` and `disableOnAction` logic

#### Copy-paste actions stop propagation
Actions of type `copy` stop event propagation by default

## 1.3.12

### Bug Fix

#### `bk-layout` correctly sets logo
Component `bk-layout` correctly sets logo URL when `userInfoUrl` property is not specified

## How to update your Console

> For on-premise Console installations, please contact your Mia-Platform referent to know how to use the `Helm chart version X.X.X`.

Author: Does this version change since we are waiting the last fixes from fast-data side?

---
id: v11.1.0
title: Version v11.1.0 Release Notes
sidebar_label: v11.1.0
image: "img/release-note-link-preview.png"
---

_May 25th, 2023_

## Console

### New Features

#### User Groups Management

> From a dedicated section inside of the Company Overview, Company Owners can now manage Company users also through Groups.  

Reviewer 4: ```suggestion
Company Owners can now manage users also through Groups from Console.  
```

Thanks to groups it is possible to simplify and speed-up the management of a multitude of users who have to be assigned the same privileges.  

> Management of users through groups sits alongside the actual individual management of users, to allow you to better structure access to resources as you prefer, giving you the freedom to choose how to govern access levels of your Company users.  

Reviewer 4: ```suggestion
Users management through groups sits alongside the current individual identities access management, to allow you to better structure access to resources and fit your teams structure, giving you the freedom to choose how to govern access levels of your Company users.  
```

![Groups section](./img/overview/group_table.png)

> To discover more about user groups management, just take a look at the related [documentation page](/development_suite/identity-and-access-management/group-management.md). 

Reviewer 4: ```suggestion
To find out more about user groups management, just take a look at the related [documentation page](/development_suite/identity-and-access-management/group-management.md). 
```

#### Privatized Marketplace for Companies

> From now on, Companies can have their own private Marketplace items, which will not be visible and usable from those who are not part of those specific Companies.   

Reviewer 4: ```suggestion
It is now possible for Companies to have their own private Marketplace items; these items will only be available to Projects within the Company, allowing for a better definition of custom resources that can speed up your development process.   
```

> More specifically, Company Owners and Project Administrators can now be able to manage Marketplace resources that will be available exclusively for the members of their Company. 

Reviewer 4: ```suggestion
Private Marketplace resources can be managed and created by Company Owners and Project Administrators. 
```

> To discover more information on how to manage Marketplace resources owned exclusively by the Company, visit the dedicated Marketplace [documentation page](/marketplace/add_to_marketplace/contributing_overview.md#public-and-private-marketplace).

Reviewer 4: ```suggestion
To find out more information on how to manage Marketplace resources owned exclusively by the Company, visit the dedicated Marketplace [documentation page](/marketplace/add_to_marketplace/contributing_overview.md#public-and-private-marketplace).
```

### Improvements

#### Check of Certificate Authority in cluster connection

> In Cluster creation, adding the Certificate Authority is actually not mandatory but it depends on the properties of each Cluster. It will always be checked: if the cluster needs a CA it will fail the connection if it has not been entered.

Reviewer 4: ```suggestion
During Cluster creation, adding the Certificate Authority is no more mandatory, based on your cluster setup you may still be required to provide a CA. Before connecting the Cluster to your Company, the Console will make sure that it can correctly reach the Cluster with a secure connection, meaning that if your Cluster uses self-signed certificate the CA will still be required.
```
